pandas/_config/config.py

```diff
@@ -827,13 +827,9 @@ def is_nonnegative_int(value: Optional[int]) -> None:
     ValueError
         When the value is not None or is a negative integer
     """
-    if value is None:
+    if isinstance(value, int) and value >= 0 or value is None:
```

Comment:
i find the status quo slightly clearer, also nice to be able to see in codecov which conditions are reached